File: src/main/java/com/alpha/licensemanagementsystem/licensemanagementsystem/model/Subscription.java
package com.alpha.licensemanagementsystem.licensemanagementsystem.model;


import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Date;


@Data
@Entity
public class Subscription implements Serializable {

    @Id @GeneratedValue(strategy= GenerationType.AUTO)
    private Long ID;

    private Date start;

    private Date end;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "plan_id")
    private Plan plan;

    @ManyToOne
    @JoinColumn(name = "region_id")
    private Region region;
}

File: frontend/src/app/Resource/Subscription.ts
import { Resource } from 'angular4-hal';
import { Plan } from './Plan';
import { User } from './User';
import { Region } from './Region';

export class Subscription extends Resource{
    start: Date
    end: Date
    id: Number
    user: User
    plan: Plan
    region: Region
}